### PR TITLE
Move route address validation from AccountPage into routes

### PR DESCRIPTION
**Pull request by lukaw3d**
Related to https://github.com/oasisprotocol/oasis-wallet-web/pull/1004#discussion_r979070318
and https://github.com/oasisprotocol/oasis-wallet-web/pull/1079#discussion_r1003929246

### CODE REVIEW

File: src/app/pages/AccountPage/index.tsx [hunk 1]
```diff
@@ -210,15 +211,8 @@ function AccountPageInternal(props: AccountPageProps) {
...
-export function AccountPage(props: AccountPageProps) {
-  const { t } = useTranslation()
-  const { address } = useParams<keyof AccountPageParams>()
-  if (!isValidAddress(address!)) {
-    return (
-      <Box pad={'small'}>
-        <AlertBox color={'status-error'}>{t('errors.invalidAddress', 'Invalid address')}</AlertBox>
-      </Box>
-    )
+export async function validateRoute(params: AccountPageParams) {
```
Comment (buberdds) [inline src/app/pages/AccountPage/index.tsx:214]: why it's async?

Comment (lukaw3d) [reply src/app/pages/AccountPage/index.tsx:214]: leftover

File: src/commonRoutes.tsx [hunk 2]
```diff
@@ -29,6 +25,10 @@ export const commonRoutes: Route[] = [
   {
     path: 'account/:address/*',
     element: <AccountPage />,
+    errorElement: <ErrorBoundary></ErrorBoundary>,
+    loader: async ({ params }) => {
```
Comment (buberdds) [inline src/commonRoutes.tsx:29]: async not needed anymore

**Review by buberdds — APPROVED** [review]

File: src/commonRoutes.tsx (hunk 2 above)
Comment (lukaw3d) [reply src/commonRoutes.tsx:29]: Oddly it works that way too, but all documentation and router code looks like it intends loader to return a Response or Promise. I'd keep it

**Pull request merged**